services/institution_service.py

```diff
@@ -0,0 +1,49 @@
+import endpoints
+from google.appengine.ext.ndb import Cursor
+
+from domain.institution_domain import Institution
+from messages.institution_messages import InstitutionListResponse
+
+
+class InstitutionService(object):
+    __instance = None
+
+    def __init__(self):
+        if InstitutionService.__instance is not None:
+            raise Exception("This class is a Singleton")
+        else:
+            InstitutionService.__instance = self
+
+    @classmethod
+    def get_instance(cls):
+        if InstitutionService.__instance is None:
+            InstitutionService()
+        return InstitutionService.__instance
+
+    def create(self, new_institution):
+        self._validate_entity_creation(new_institution)
+        return self._save(new_institution)
+
+    def list_entities_by_code(self, request):
+        cursor = Cursor(urlsafe=request.cursor)
+        query = Institution.query()
+        print"\n\n\n"
+        print request
+        if request.code:
+            query = Institution.query(Institution.code == request.code)
+        institutions, next_cursor, more = query.fetch_page(page_size=request.page_size, start_cursor=cursor)
+        response = InstitutionListResponse()
+        response.institutions = [Institution.encode(i) for i in institutions]
+        if more and next_cursor:
+            response.cursor = next_cursor.urlsafe()
+        return response
+
+    def _validate_entity_creation(self, new_institution):
+        pass
+        # raise endpoints.BadRequestException("Error while creating Institution")
```

Comment:
I'm pretty sure this is a TODO item, but am flagging it just to make sure it's not forgotten.

Reply:
Yes, it's on my TODO list. I did not forget about it. Thanks anyway.